src/composables/accountSwitcher.ts

```diff
@@ -0,0 +1,87 @@
+import { computed, Ref } from '@vue/composition-api';
+import { useGetter, useState } from './vuex';
+import { useMultisigAccounts } from './multisigAccounts';
+
+import type {
+  IAccount,
+  IDefaultComposableOptions,
+  IMultisigAccount,
+} from '../types';
+
+interface UseAccountSwitcherOptions extends IDefaultComposableOptions {
+  isMultisigPage?: Ref<boolean>
+}
+
+interface IAccountSwitcherBase {
+  activeIdx: Ref<number>,
+  // eslint-disable-next-line no-unused-vars
+  selectAccount: (realIndex: number) => void,
+}
+
+interface IMultisigAccountSwitcher extends IAccountSwitcherBase {
+  accounts: Ref<IMultisigAccount[]>,
+  // eslint-disable-next-line no-unused-vars
+  getKeyValue: (account: IMultisigAccount) => string
+}
+
+interface IStandardAccountSwitcher extends IAccountSwitcherBase {
+  accounts: Ref<IAccount[]>,
+  // eslint-disable-next-line no-unused-vars
+  getKeyValue: (account: IAccount) => string
+}
+
+function useMultisigAccountSwitcher(
+  { store }: UseAccountSwitcherOptions,
+): IMultisigAccountSwitcher {
+  const {
+    multisigAccounts,
+    activeMultisigAccountId,
+    setActiveMultisigAccountId,
+  } = useMultisigAccounts({ store });
+
+  const multisigAccountIdx = computed(
+    () => multisigAccounts.value.findIndex((acc) => acc.address === activeMultisigAccountId.value),
+  );
+
+  function selectAccount(realIndex: number) {
+    setActiveMultisigAccountId(multisigAccounts.value[realIndex].multisigAccountId);
+  }
+
+  function getKeyValue(account: IMultisigAccount) {
+    return account.multisigAccountId;
+  }
+
+  return {
+    accounts: multisigAccounts,
+    activeIdx: multisigAccountIdx,
+    selectAccount,
+    getKeyValue,
+  };
+}
+function useStandardAccountSwitcher(
+  { store }: UseAccountSwitcherOptions,
+): IStandardAccountSwitcher {
+  const activeIdx = useState<number>('accounts', 'activeIdx');
+  const accounts = useGetter<IAccount[]>('accounts');
```

Comment:
we cannot use useState and useGetter, but wen do refactor in vuex.ts, we can resolve that problem. Let's sync up with me